src/coreclr/jit/gentree.cpp

```diff
@@ -20431,8 +20452,16 @@ GenTree* Compiler::gtNewSimdBinOpNode(
                     if (varTypeIsLong(simdBaseType) || (simdBaseType == TYP_DOUBLE))
                     {
                         assert(varTypeIsSigned(simdBaseType));
-                        assert(compIsaSupportedDebugOnly(InstructionSet_AVX512F_VL));
-                        intrinsic = NI_AVX512F_VL_ShiftRightArithmetic;
+                        assert(compIsaSupportedDebugOnly(InstructionSet_AVX512F_VL) ||
```

Comment:
There's a case missed above in `gtNewSimdBinOpNode` where `NI_AVX512DQ_And` is used and it should prefer `NI_AVX10v1_V512_And` for floating-point types.

Reply:
Similar for `NI_AVX512DQ_AndNot`

Reply:
Also similar for `NI_AVX512DQ_Or` and `NI_AVX512DQ_Xor`

Reply:
I think there was a confusion during our internal discussion. As of now, I have not added handling for any of the ```AVX10v1_V512``` nodes. Since we will always have ```Avx512F`` when we have 512 bit support, I have not changed any handling for 512 bit APIs. Let me know if I am wrong here. 

That is the reason we have following implications here
```
implication        ,X86   ,AVX10v1_V512         ,AVX512F
implication        ,X86   ,AVX10v1_V512         ,AVX512F_VL
implication        ,X86   ,AVX10v1_V512         ,AVX512CD
implication        ,X86   ,AVX10v1_V512         ,AVX512CD_VL
implication        ,X86   ,AVX10v1_V512         ,AVX512BW
implication        ,X86   ,AVX10v1_V512         ,AVX512BW_VL
implication        ,X86   ,AVX10v1_V512         ,AVX512DQ
implication        ,X86   ,AVX10v1_V512         ,AVX512DQ_VL
implication        ,X86   ,AVX10v1_V512         ,AVX512VBMI
implication        ,X86   ,AVX10v1_V512         ,AVX512VBMI_VL
```

And hence we decided not to add extra code for handling any of the above.

Reply:
@tannergooding Do you mean changing `NI_AVX512DQ_And` to `AVX10v1_V512_And` here(and in similar places) : https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/gentree.cpp#L20236

So this would look like 

```
.........
case GT_AND:
        {
            if (simdSize == 64)
            {
                assert(compIsaSupportedDebugOnly(InstructionSet_AVX512F));
                intrinsic = NI_AVX512F_And;

                if (varTypeIsIntegral(simdBaseType))
                {
                    intrinsic = NI_AVX512F_And;
                }
                else if (compOpportunisticallyDependsOn(InstructionSet_AVX512DQ))
                {
                    intrinsic = AVX10v1_V512_And;
                }
                else
                {
                    // Since this is a bitwise operation, we can still support it by lying
                    // about the type and doing the operation using a supported instruction

                    intrinsic       = NI_AVX512F_And;
                    simdBaseJitType = (simdBaseType == TYP_DOUBLE) ? CORINFO_TYPE_LONG : CORINFO_TYPE_INT;
                }
            }
............................
```

Reply:
Ah, you're right. I was thinking it was needed but `Avx10v1_V512` implies almost all `Avx512*` ISAs and so we don't actually need to do anything. It's only the `_VL` ones that need to ensure we're handling both paths.

My mistake

Reply:
Thanks Tanner and Deepak. So I will consider all the similar comments resolved.